tests/integration/cattletest/core/test_authorization.py

```diff
@@ -1,6 +1,23 @@
 from common_fixtures import *  # NOQA
 
 
+def test_client_access(clients):
+    typesLen = {
+        'admin': 89,
+        'agent': 8,
+        'user': 68,
+        'agentRegister': 4,
+        'test': 140,
+        'readAdmin': 89,
+        'token': 2,
+        'superadmin': 140,
+        'service': 89,
+        'project': 68,
+    }
+    for tuple in clients.items():
+        assert typesLen[tuple[0]] == len(tuple[1].schema.types.items())
```

Comment:
Nah.  Don't worry about it.  This is fine.

On Sun, Apr 19, 2015 at 10:42 PM, James Harris <notifications@github.com>
wrote:

> In tests/integration/cattletest/core/test_authorization.py
> <https://github.com/rancherio/cattle/pull/313#discussion_r28665027>:
>
> > @@ -1,6 +1,23 @@
> >  from common_fixtures import *  # NOQA
> >
> >
> > +def test_client_access(clients):
> > +    typesLen = {
> > +        'admin': 89,
> > +        'agent': 8,
> > +        'user': 68,
> > +        'agentRegister': 4,
> > +        'test': 140,
> > +        'readAdmin': 89,
> > +        'token': 2,
> > +        'superadmin': 140,
> > +        'service': 89,
> > +        'project': 68,
> > +    }
> > +    for tuple in clients.items():
> > +        assert typesLen[tuple[0]] == len(tuple[1].schema.types.items())
>
> Drone will tell u which schema has the wrong length. But when that is
> corrected then it will more than likely occur again. I could rewrite this
> so that it creates two sets that are compared and it would spit both sets
> out that way you can see it in one failure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rancherio/cattle/pull/313/files#r28665027>.
>